src/ucp/core/ucp_context.c

```diff
@@ -666,6 +679,12 @@ static ucs_status_t ucp_fill_resources(ucp_context_h context,
          * client-server connection establishment via sockaddr, don't use it */
         if ((num_tl_resources > 0) ||
             (context->tl_mds[md_index].attr.cap.flags & UCT_MD_FLAG_SOCKADDR)) {
+            /*List of memory type MDs*/
+            if (!(mem_type_mask & UCS_BIT(context->tl_mds[md_index].attr.cap.mem_type))) {
```

Comment:
can cache the value of UCS_BIT(context->tl_mds[md_index].attr.cap.mem_type) in a variable, it's used twice, will make the code a bit easier to read